### PR TITLE
[New] Add vector tiled layer from custom style

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR implements `Add vector tiled layer from custom style` in `SAMPLE_CATEGORY` category.
URL to README: [Add vector tiled layer from custom style](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-AddVectorTiledLayerFromCustomStyle/Shared/Samples/Add%20vector%20tiled%20layer%20from%20custom%20style)

## Linked Issue(s)

- `swift/issues/6217`

## How To Test

- Use the picker to change the style of the vector tiled basemape.

## Screenshots

add-vector-tiled-layer-from-custom-style-1    add-vector-tiled-layer-from-custom-style-2

### CODE REVIEW

File: Shared/Samples/Add vector tiled layer from custom style/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Add vector tiled layer from custom style
+
+Load an ArcGIS vector tiled layers using custom styles.
+
+![Screenshot of Add vector tiled layer from custom style sample](add-vector-tiled-layer-from-custom-style-1.png)
+![Screenshot of Add vector tiled layer from custom style sample](add-vector-tiled-layer-from-custom-style-2.png)
```
Comment (philium) [inline Shared/Samples/Add vector tiled layer from custom style/README.md:6]: These images both have the same description. Consider differentiating them.

File: Shared/Samples/Add vector tiled layer from custom style/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+1. Construct an `ArcGISVectorTiledLayer` with the URL of a custom style from AcrGIS Online.
+2. Alternatively, construct an `ArcGISVectorTiledLayer` by taking a portal item offline and apply it to an offline vector tile package:
+    i. Create a `PortalItem` using the URL of a custom style.  
+    ii. Create an `ExportVectorTilesTask` using the portal item.  
+    iii. Get the `ExportVectorTilesJob` using `ExportVectorTilesTask.makeExportStyleResourceCacheJob(itemResourceCacheURL:)`.  
+    iv. Start the job using  `ExportVectorTilesJob.start()`.  
+    v. Construct an `VectorTileCache` using the path of the local vector tile package.  
+    vi. Once the job is complete, construct an `ArcGISVectorTiledLayer` using the vector tile cache and the `ItemResourceCache` from the job's result.  
+3. Create an `Basemap` from the `ArcGISVectorTiledLayer`.
+4. Assign the basemap to the map's `basemap`.
```
Comment (philium) [inline Shared/Samples/Add vector tiled layer from custom style/README.md:27]: Swift doesn't have constructors.

```suggestion
1. Create an `ArcGISVectorTiledLayer` with the URL of a custom style from AcrGIS Online.
2. Alternatively, create an `ArcGISVectorTiledLayer` by taking a portal item offline and applying it to an offline vector tile package:
    i. Create a `PortalItem` using the URL of a custom style.  
    ii. Create an `ExportVectorTilesTask` using the portal item.  
    iii. Get the `ExportVectorTilesJob` using `ExportVectorTilesTask.makeExportStyleResourceCacheJob(itemResourceCacheURL:)`.  
    iv. Start the job using  `Job.start()`.  
    v. Create a `VectorTileCache` using the path of the local vector tile package.  
    vi. Once the job is complete, create an `ArcGISVectorTiledLayer` using the vector tile cache and the `ItemResourceCache` from the job's result.  
3. Create a `Basemap` from the `ArcGISVectorTiledLayer`.
4. Assign the basemap to the map's `basemap`.
```

File: Shared/Samples/Add vector tiled layer from custom style/AddVectorTiledLayerFromCustomStyleView.swift [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            }
+            map.basemap = Basemap(baseLayer: vectorTiledLayer)
+            
+            if vectorTiledLayer.vectorTileCache != nil {
+                // Uses a Dodge City, KS viewpoint if the layer was created using the tile cache.
+                return Viewpoint(latitude: 37.76528, longitude: -100.01766, scale: 4e4)
+            } else {
+                // Uses a Europe/Africa viewpoint if the layer was created using an online style.
+                return Viewpoint(latitude: 28.53345, longitude: 17.56488, scale: 1e8)
+            }
```
Comment (philium) [inline Shared/Samples/Add vector tiled layer from custom style/AddVectorTiledLayerFromCustomStyleView.swift:118]:
```suggestion
            return if vectorTiledLayer.vectorTileCache != nil {
                // Uses a Dodge City, KS viewpoint if the layer was created using the tile cache.
                Viewpoint(latitude: 37.76528, longitude: -100.01766, scale: 4e4)
            } else {
                // Uses a Europe/Africa viewpoint if the layer was created using an online style.
                Viewpoint(latitude: 28.53345, longitude: 17.56488, scale: 1e8)
            }
```

**Review by philium — APPROVED** [review]

**Review by dfeinzimer — APPROVED** [review]

**Comment by CalebRas** [comment]
Thank you for looking at this!

**Pull request merged**